articles/iot-hub/iot-hub-devguide-messages-d2c.md

```diff
@@ -119,7 +119,7 @@ You can enable/disable the fallback route in the Azure portal->Message Routing b
 
 ## Non-telemetry events
 
-In addition to device telemetry, message routing also enables sending device twin change events, device lifecycle events, digital twin change events, and device connection state events. For example, if a route is created with data source set to **device twin change events**, IoT Hub sends messages to the endpoint that contain the change in the device twin. Similarly, if a route is created with data source set to **device lifecycle events**, IoT Hub sends a message indicating whether the device was deleted or created. As part of [Azure IoT Plug and Play](../iot-develop/overview-iot-plug-and-play.md), a developer can create routes with data source set to **digital twin change events** and IoT Hub sends messages whenever a digital twin property is set or changed, a digital twin is replaced, or when a change event happens for the underlying device twin. Finally, if a route is created with data source set to **device connection state events**, IoT Hub sends a message indicating whether the device was connected or disconnected.
+In addition to device telemetry, message routing also enables sending device twin change events, device lifecycle events, digital twin change events, and device connection state events. For example, if a route is created with data source set to **device twin change events**, IoT Hub sends messages to the endpoint that contain the change in the device twin. Similarly, if a route is created with data source set to **device lifecycle events**, IoT Hub sends a message indicating whether the device or module was deleted or created. More details on device lifecycle events can be found [here](./iot-hub-devguide-identity-registry.md#device-and-module-lifecycle-notifications). As part of [Azure IoT Plug and Play](../iot-develop/overview-iot-plug-and-play.md), a developer can create routes with data source set to **digital twin change events** and IoT Hub sends messages whenever a digital twin property is set or changed, a digital twin is replaced, or when a change event happens for the underlying device twin. Finally, if a route is created with data source set to **device connection state events**, IoT Hub sends a message indicating whether the device was connected or disconnected.
```

Comment:
A few things:

1. You can't use "here: as link text: "More details on device lifecycle events can be found [here](./iot-hub-devguide-identity-registry.md#device-and-module-lifecycle-notifications)." Please change this to: " For more information about device lifecycle events, see [Device and module lifecycle notifications](./iot-hub-devguide-identity-registry.md#device-and-module-lifecycle-notifications)." or something to that effect. You could also just say: "For more information, see ..." since the link text is pretty self-descriptive.

2. Instead of "As part of [IoT Plug and Play]", I'd probably say "When using [IoT Plug and Play]" -- that's more of a nit. 

3. One other thing. I'm not sure if it's covered elsewhere in the article, but you might want to mention that connection events are not sent for devices that are using HTTPS, I know this is the case for Event Grid, I'll bet it's also the case here, but double check. (We had someone comment on this issue when they were trying to get connection events through event grid.